### PR TITLE
Add waiting and erroring resources

### PR DESCRIPTION
**Pull request by randhid**
Added new types of fakes to some models, `erroring` - return errors everywhere and `waiting` - wait for a configured time before returning their rpcs. The second is to test out code behaviors in Olivia's failover components project. Asked for both of you to take a look before I uploaded. Verified it can build and does not hang on startup or shutdown.

### CODE REVIEW

**Review by oliviamiller — APPROVED** [review]
waiting components lgtm! thanks for adding

File: common/common.go [hunk 1]
```diff
@@ -3,11 +3,16 @@ package common
...
+	"time"
 
 	"go.viam.com/rdk/resource"
 )
 
-const GoalWithinRange = 0.2
+const (
+	GoalWithinRange   = 0.2
+	DefaultWaitTimeMs = 500 * time.Millisecond // default wait time in milllisecods
+
```
Comment (randhid) [inline common/common.go]: space

File: fakemovementsensor/waiting.go [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	}
+
+	if err := w.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+
+	return w, nil
+}
+
+func (w *waiting) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
```
Comment (randhid) [inline fakemovementsensor/waiting.go]:
```suggestion
func (w *waiting) Reconfigure(_ context.Context, _ resource.Dependencies, conf resource.Config) error {
```

File: fakepowersensor/waiting.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+	resource.AlwaysRebuild
+	resource.TriviallyCloseable
+	logger logging.Logger
+
+	mu       sync.Mutex
+	ac       bool
+	waitTime time.Duration
+}
+
+func newWaitingPowerSensor(_ context.Context, _ resource.Dependencies, conf resource.Config, logger logging.Logger) (
```
Comment (randhid) [inline fakepowersensor/waiting.go:34]: call new fake from within driver and wait before returning - cleaner

File: fakesensor/erroring.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	e := &erroring{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+
+	return e, nil
+}
+
+func (e *erroring) Readings(context.Context, map[string]interface{}) (map[string]interface{}, error) {
+	return nil, nil
```
Comment (randhid) [inline fakesensor/erroring.go]:
```suggestion
	return nil, grpc.Unimplemented
```

File: fakepowersensor/waiting.go (hunk 3 above)
Comment (randhid) [reply fakepowersensor/waiting.go:34]: won't do now, but can in future.

**Review by zaporter-work — COMMENTED** [review]
Didn't see this in time, but LGTM

File: meta.json [hunk 5]
```diff
@@ -115,7 +143,7 @@
   ],
   "build": {
     "build": "make --always-make module.tar.gz",
-    "arch" : ["linux/amd64", "linux/arm64", "darwin/arm64"]
```
Comment (zaporter-work) [inline meta.json:118]: I got it building on linux. Happy to help fix this for linux builds

**Pull request merged**